tests/console/verify_systemd_timesync.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that systemd timer is used for time synchronization.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Check yast-timesync.timer", "Check if file /etc/systemd/system/yast-timesync.timer exists and has expected configuration.");
+    assert_script_run('ls /etc/systemd/system/yast-timesync.timer');
+    my $conf_file = script_output("cat /etc/systemd/system/yast-timesync.timer");
+    $conf_file =~ /OnUnitActiveSec=(?<interval>(\S+))min/;
+    my $expected_interval = $test_data->{profile}->{'ntp-client'}->{ntp_sync};
+    assert_equals($expected_interval, $+{interval}, "The interval in yast-timesync configuration file is not the expected one.");
+    # Default value for first time sync is 1 minute after boot. Module pauses in order to be able to track the sync in the logs.
+    sleep 60;
+    record_info("Check logs for timesync", "Check if the use of systemd timers can be spotted in the journal logs as it's expected.");
+    assert_script_run("journalctl -u yast-timesync | grep \"Started One time sync configured by YaST\"");
+    record_info("Check time deviation correction", "Check if \"One time synchronization\" will fix time deviation as expected.");
+    script_run("date");
+    assert_script_run('date $(date +%m%d%H%M%Y.%S -d "1 minute ago")');
+    my $time_deviation;
+    my $tries = 0;
+    while (($time_deviation < 59) && ($tries < 100)) {
```

Comment:
let me know if I'm missing something here, but if we know [last sync](https://openqa.suse.de/tests/4103972#step/verify_systemd_timesync/14) 14:06:56, we know in this scenario that at 14:9:56 will be another sync with that expected message, we can just calculate exact number of seconds and check. I would go with 1 sec second instead of 3 to make the test module run less time. wdyt?

Regarding current loop I think we would be running some instruction without need plus trusting the command to run in 3 seconds or so? which I guess it would depend on the processor. Also it could be some unexpected behavior if takes more that 100 tries comparing `undef > 59`. In this repo as far as I know when trying to do some loop to wait, active wait, there is always a sleep inside the while to wait and retry.

Reply:
Yes, I can reduce interval to 1 minute now (I cannot set interval in seconds). I had it set to 3 min because of my previous approach, in order to avoid chances of timesync happening while typing commands. Command takes 15 seconds to run, I can add a sleep for interval*60 - 20 so it runs only once in every interval, is that what you mean?

Reply:
This can't work. See comment above.